src/controllers/beneficial.owner.other.controller.ts

```diff
@@ -1,29 +1,44 @@
 import { NextFunction, Request, Response } from "express";
-import { BENEFICIAL_OWNER_OTHER_PAGE, BENEFICIAL_OWNER_TYPE_URL } from "../config";
+import {
+  BENEFICIAL_OWNER_OTHER_PAGE,
+  BENEFICIAL_OWNER_TYPE_URL,
+  FEATURE_FLAG_ENABLE_REDIS_REMOVAL,
+  BENEFICIAL_OWNER_TYPE_WITH_PARAMS_URL
+} from "../config";
+import { isActiveFeature } from "../utils/feature.flag";
 import {
   getBeneficialOwnerOther,
   getBeneficialOwnerOtherById,
   postBeneficialOwnerOther,
   removeBeneficialOwnerOther,
   updateBeneficialOwnerOther
 } from "../utils/beneficial.owner.other";
+import { getUrlWithParamsToPath } from "../utils/url";
 
 export const get = (req: Request, res: Response) => {
-  getBeneficialOwnerOther(req, res, BENEFICIAL_OWNER_OTHER_PAGE, BENEFICIAL_OWNER_TYPE_URL);
+  getBeneficialOwnerOther(req, res, BENEFICIAL_OWNER_OTHER_PAGE, nextOrLastPage(req));
 };
 
 export const getById = (req: Request, res: Response, next: NextFunction) => {
-  getBeneficialOwnerOtherById(req, res, next, BENEFICIAL_OWNER_OTHER_PAGE, BENEFICIAL_OWNER_TYPE_URL);
+  getBeneficialOwnerOtherById(req, res, next, BENEFICIAL_OWNER_OTHER_PAGE, nextOrLastPage(req));
 };
 
 export const post = (req: Request, res: Response, next: NextFunction) => {
-  postBeneficialOwnerOther(req, res, next, BENEFICIAL_OWNER_TYPE_URL, true);
+  postBeneficialOwnerOther(req, res, next, nextOrLastPage(req), true);
 };
 
 export const update = (req: Request, res: Response, next: NextFunction) => {
-  updateBeneficialOwnerOther(req, res, next, BENEFICIAL_OWNER_TYPE_URL, true);
+  updateBeneficialOwnerOther(req, res, next, nextOrLastPage(req), true);
 };
 
 export const remove = (req: Request, res: Response, next: NextFunction) => {
-  removeBeneficialOwnerOther(req, res, next, BENEFICIAL_OWNER_TYPE_URL, true);
+  removeBeneficialOwnerOther(req, res, next, nextOrLastPage(req), true);
+};
+
+const nextOrLastPage = (req: Request): string => {
+  let nextPage = BENEFICIAL_OWNER_TYPE_URL;
```

Comment:
I agree with Martin about the function name, perhaps getBeneficialOwnerTypeUrl() would be better? it would avoid the not knowing if it was in fact a next page or previous page.

Reply:
Updated the name as suggested.